R/categorical_trees.R

```diff
@@ -701,11 +701,7 @@ check_agg_scale_subtree_dt <- function(dt,
   # combine actual and expected datasets
   dt[, data_exists := TRUE]
   diagnostic_id_cols <- c(by_id_cols, col_stem)
-  diagnostic_dt <- merge(
-    dt, expected_dt,
-    all = T,
-    by = diagnostic_id_cols
-  )
+  diagnostic_dt <- dt[expected_dt, on = diagnostic_id_cols]
```

Comment:
This section checks that all expected rows in `dt` exist. `profvis` identified this merge as taking about ~19 seconds for the largest aggregation to all-ages. Switching to the data.table join reduced it to about ~4 seconds. 

This speed difference is explained in an upcoming data.table vignette https://github.com/zeomal/data.table/blob/patch-1/vignettes/datatable-joins.Rmd#L293

Reply:
Does this need to be done by merge?